src/project.cpp

```diff
@@ -0,0 +1,120 @@
+/* ****************************************************************************
+*
+* Copyright (c) Microsoft Corporation. All rights reserved.
+*
+*
+* This file is part of Microsoft R Host.
+*
+* Microsoft R Host is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Microsoft R Host is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Microsoft R Host.  If not, see <http://www.gnu.org/licenses/>.
+*
+* ***************************************************************************/
+
+
+#include "project.h"
+#include "util.h"
+#include <zip.h>
+
+namespace fs = std::experimental::filesystem;
+
+namespace rhost {
+    namespace rproj {
+        namespace {
+            void extract_project_internal(fs::path zip_file, fs::path dest_dir, fs::path temp_dir) {
+                // Open ZIP archive file
+                int zip_err = ZIP_ER_OK;
+                zip_t* archive = zip_open(zip_file.make_preferred().string().c_str(), ZIP_RDONLY, &zip_err);
+                SCOPE_WARDEN(zip_close, {
+                    if (archive) {
+                        zip_close(archive);
+                    }
+                });
+
+                if (zip_err != ZIP_ER_OK) {
+                    throw std::exception("Error while opening compressed project file.");
+                }
+
+                // Get the number of entries in the archive
+                zip_int64_t nfiles = zip_get_num_entries(archive, ZIP_FL_UNCHANGED);
+
+                // 10 MB buffer to extract files
+                size_t buff_size = 10 * 1024 * 1024;
+                std::unique_ptr<byte[]> buffer(new byte[buff_size]);
```

Comment:
Where does `byte` come from? I suspect it's in Win32 headers somewhere, and we shouldn't depend on those where we don't have to. Just using `char` directly is fine - there's nothing useful that `typedef` actually adds here, since `sizeof(char)` is guaranteed to be 1.